src/packet/outgoing.rs

```diff
@@ -0,0 +1,202 @@
+use crate::{
+    net::constants::{DEFAULT_ORDERING_STREAM, DEFAULT_SEQUENCING_STREAM},
+    packet::{
+        header::{
+            AckedPacketHeader, ArrangingHeader, FragmentHeader, HeaderWriter, StandardHeader,
+        },
+        DeliveryGuarantee, OrderingGuarantee, PacketType,
+    },
+};
+
+/// Builder that could be used to construct an outgoing laminar packet.
+pub struct OutgoingPacketBuilder<'p> {
+    header: Vec<u8>,
+    payload: &'p [u8],
+}
+
+impl<'p> OutgoingPacketBuilder<'p> {
```

Comment:
Should we be using expect in this section, since it will panic? Why not return an Result?

Reply:
I did this for two reasons:

1.  When using a builder and returning `Self` we can't use the API properly because you need  to unwrap:

```
buidler.with_fragment_header()?.with_default_header()?.with_sequencing_header()?;
````

2. If the parsing of our headers is wrong, I don't see any reason to continue with receiving the packet. We can choose to stop handling this packet. However, parsing code and invalid headers are probably an issue of our code and it'll probably be failing for the next packet too. It could be possible that send data is invalid, but we have most of this covered with the CRC32, although one can construct a packet with the right CRC32 and create its own packet.

Now I think of this we should probably have a CRC32 over the whole packet content too.

Reply:
TCP handles its own fragmentation. We shouldn't be adding anything on top of TCP, I would think...

Reply:
The CRC32 should definitely include the payload. Otherwise, it isn't really worth having to be honest.

Reply:
Yea, the idea is that the CRC32 is smaller than a string for a header. We need to check on the protocol version, I think that CRC32 is fitting this purpose well. 

However, we probably want to have a CRC32 over the whole payload. I will create an issue for this.

Reply:
https://github.com/amethyst/laminar/issues/151